### DISCUSSION TITLE
Change in histogram(grains) function?

### DISCUSSION THREAD

**Discussion by S-Tholen** [mtex-toolbox/mtex#541] [Ask Anything] [answered]
So far I used (from time to time) the histogram(grains) function as described in https://mtex-toolbox.github.io/ShapeParameters.html.

Now I got an error message (see below) for both, my own dataset and the mtex forsterite one.

Plotting properties (histogram(grains.grainSize) etc.) still works fine...

Anything changed?

***
Error using histogram
Expected input number 1, x, to be one of these types:

double, single, uint8, uint16, uint32, uint64, int8, int16, int32, int64, logical, datetime, duration

Instead its type was grain2d.

Error in histogram>parseinput (line 257)
        validateattributes(x,{'numeric','logical','datetime','duration'},...

Error in histogram (line 145)
[opts,passthrough,dispatchToCategorical] = parseinput(args,firstaxesinput);
***

**Comment by kilir** [comment]
Hi,
using the current Mtex version (development), I get no issues.
What version of Mtex are you using?

Cheers,
Rüdiger

**Comment by ralfHielscher** [comment, marked as answer]
Hi,

`histogram(grains)` works only in the current beta. You can still use `hist(grains)`

Ralf.

**Reply by S-Tholen** [reply]
hist(grains) works fine. Thanks alot!
